### PR TITLE
Add capability to set surface export values via data from file(s)

### PR DESCRIPTION
**Pull request by AaronDonahue**
This commit will take advantage of the new TimeInterpolation tool introduced in #2338 to allow users to specify surface export fields that will be set by data found in a set of files.  This is a runtime option controlled in a way similar to prescribing certain export variables to be set to a constant.

This commit augments the surface_coupling stand alone test to apply from_file exports for an export field.

### CODE REVIEW

**Comment by AaronDonahue** [comment]
WIP pending #2238

**Comment by AaronDonahue** [comment]
Noting here that in addition to the unit test this PR introduces, @elynnwu and @AaronDonahue independently checked the surface coupling variables from ATM when using the prescribed from file option against the intended prescribed values on a F-Case and they matched.  So another layer of confirmation that this function acts as expected.

**Comment by AaronDonahue** [comment]
The fails in the AT are because I was using a prebuilt `.nc` file for the surface coupling test, but we have set those to be ignored by git.  I'm working on adding code to build the file on the fly instead.

**Comment by AaronDonahue** [comment]
> The fails in the AT are because I was using a prebuilt `.nc` file for the surface coupling test, but we have set those to be ignored by git. I'm working on adding code to build the file on the fly instead.

Okay, it should be fixed now

**Review by tcclevenger — COMMENTED** [review]
The surface coupling test has an input.yaml and runs through the AD, would it make sense to have a `prescribed_from_file` entry  for that test? Same with `prescribed_constants`. To make sure the parameters are parsed correctly?

File: components/eamxx/src/control/atmosphere_surface_coupling_exporter.hpp [hunk 1]
```diff
@@ -137,7 +143,8 @@ class SurfaceCouplingExporter : public AtmosphereProcess
   uview_2d<HostDevice,    Real> m_cpl_exports_view_h;
 
   // Array storing the field names for exports
-  name_t* m_export_field_names;
+  name_t*                   m_export_field_names;
+  std::vector<std::string>  m_export_field_names_vector;
```
[the remark this answers is not in the document]
Comment (AaronDonahue) [reply components/eamxx/src/control/atmosphere_surface_coupling_exporter.hpp:147]: I needed a vector representation of `m_export_field_names` so I can have a list (in this case vector) of names that was searchable w/ find.  Is there a way to search a array of characters?

[the remark this answers is not in the document]
Comment (tcclevenger) [reply components/eamxx/src/control/atmosphere_surface_coupling_exporter.hpp:147]: Sorry, I misread this. I was commenting on the name, but now I see there is still a `m_export_field_names` and `m_export_field_names_vector`. So ignore what I said.

[the remark this answers is not in the document]
Comment (bartgol) [reply components/eamxx/src/control/atmosphere_surface_coupling_exporter.hpp:147]: I find it confusing to have these two members though. Can we just have the std::vector one?

[the remark this answers is not in the document]
Comment (bartgol) [reply components/eamxx/src/control/atmosphere_surface_coupling_exporter.hpp:147]: I'm actually wondering if we should avoid the manual pointers completely in the SC stuff. But I don't want to hijack this PR...

[the remark this answers is not in the document]
Comment (tcclevenger) [reply components/eamxx/src/control/atmosphere_surface_coupling_exporter.hpp:147]: I agree. How important do we think this is? I have some free cycles currently.

File: components/eamxx/tests/uncoupled/surface_coupling/surface_coupling.cpp [hunk 2]
```diff
@@ -16,6 +17,112 @@
...
+// Test function for prescribed values
+Real test_func(const int col, const int t) {
+  return (Real)(col + 1 + t);
+}
+
+// Wrapper for output manager that also extracts the list of files
+class OutputManager4Test : public scream::OutputManager
+{
+public:
+  OutputManager4Test()
```
Comment (bartgol) [inline components/eamxx/tests/uncoupled/surface_coupling/surface_coupling.cpp:31]: Minor, but `OutputManager4Test () = default` conveys more intent (and it's shorter)

File: components/eamxx/tests/uncoupled/surface_coupling/surface_coupling.cpp [hunk 3]
```diff
@@ -16,6 +17,112 @@
 
 namespace scream {
 
+constexpr Real test_tol = std::numeric_limits<Real>::epsilon()*1e4;
+
+// Test function for prescribed values
+Real test_func(const int col, const int t) {
+  return (Real)(col + 1 + t);
```
Comment (bartgol) [inline components/eamxx/tests/uncoupled/surface_coupling/surface_coupling.cpp:24]: Minor: the (C style) cast is not needed. The compiler already does that for you.

**Review by bartgol — APPROVED** [review]

**Review by tcclevenger — APPROVED** [review]

**Review by tcclevenger — APPROVED** [review]

**Review by bartgol — DISMISSED** [review]

**Comment by AaronDonahue** [comment]
@bartgol and @tcclevenger can you take a look at the latest changes to the PR that I added to get it in line with master?

**Review by tcclevenger — APPROVED** [review]

**Pull request merged**